application-licensing-licensor/application-licensing-licensor-api/src/main/resources/templates/extensionAutoUpgraded/extensionAutoUpgraded.vm

```diff
@@ -0,0 +1,53 @@
+#template('notification/macros.vm')
+
+#includeMacros('Licenses.Code.VelocityMacros')
+
+#set ($mainIcon = "bell")
+#set ($smallIcon = "arrow_up")
+#set ($compositeEventDate = $escapetool.xml($datetool.whenIs($event.dates.get(0))))
+
+#macro (displayEventDetails $event)
+  #set ($userName = $event.user.name)
+  <tr>
+    <td>
+      <span class="notification-event-user">$userName</span>
+    </td>
+    <td class="description">$event.getBody()</td>
```

Comment:
I assume the event body can have HTML?

Reply:
The event body is just a  simple text message added  https://github.com/xwikisas/application-licensing/blob/master/application-licensing-licensor/application-licensing-licensor-api/src/main/java/com/xwiki/licensing/internal/upgrades/notifications/ExtensionAutoUpgradedEventConverter.java#L61
which is constructed in here https://github.com/xwikisas/application-licensing/blob/master/application-licensing-licensor/application-licensing-licensor-api/src/main/java/com/xwiki/licensing/internal/upgrades/UpgradeExtensionHandler.java#L118

Reply:
So it seems it doesn't contain HTML, at least not ATM, thus it would be safer I think, for now, to escape the special XML characters. Users can mess with translations, putting values we don't control, but this shouldn't break the XWiki UI.